### PR TITLE
Implement SetTimberUserContext plug

### LINKED ISSUE #932: Move Timber context adding out of GuardianSerializer into a Plug
# Problem

Right now, Timber logging user context is being added as a side effect of deserializing a user from a token in `CodeCorpsWeb.Plugs.GuardianSerializer`.

Being an unexpected side effect of indirectly related code, this is a code smell and should be moved out of the token and into a plug of it's own.

### PR DESCRIPTION
**Pull request by tteerawat**
# What's in this PR?

- Create `CodeCorpsWeb.Plug.SetTimberUserContext` plug module.
- Implement `User.full_name/1` function.
- Remove unused code from `CodeCorpsWeb.GuardianSerializer` module.
- Apply `CodeCorpsWeb.Plug.SetTimberUserContext` module to:
  - Router file in `:current_user` pipeline.

## References
Fixes #932

### CODE REVIEW

**Comment by joshsmith** [comment]
@begedin would you mind giving this a look?

**Review by begedin — DISMISSED** [review]
Well tested and more explicit than what we have. Looks good to me.

**Comment by joshsmith** [comment]
There are some conflicts here now, though. We've upgraded Guardian.

@teerawat1992 is it possible for you to deconflict the changes and bring them into the newly updated Guardian code?

**Comment by tteerawat** [comment]
@joshsmith Sure, I'll deconflict the changes 🙇

**Comment by tteerawat** [comment]
@joshsmith @begedin I've rebased from develop branch and refactored `Guardian` module, please review again when you guys have time 🙇

File: test/lib/code_corps/model/user_test.exs [hunk 1]
```diff
@@ -6,20 +6,39 @@ defmodule CodeCorps.UserTest do
   @valid_attrs %{email: "test@user.com", password: "somepassword", username: "testuser"}
   @invalid_attrs %{}
 
-  test "changeset with valid attributes" do
-    changeset = User.changeset(%User{}, @valid_attrs)
-    assert changeset.valid?
+  describe "full_name/1" do
+    test "returns full name correclty" do
```
Comment (begedin) [inline test/lib/code_corps/model/user_test.exs]: got a typo here, sorry 😬

**Review by begedin — CHANGES_REQUESTED** [review]
Once the typo is cleared, I'm fine with merging this in.

File: test/lib/code_corps/model/user_test.exs (hunk 1 above)
Comment (tteerawat) [reply test/lib/code_corps/model/user_test.exs]: whoops 😅

**Review by begedin — APPROVED** [review]

**Comment by joshsmith** [comment]
🙌  great work @teerawat1992!

**Pull request merged**